tests/rbd_mirror/test-add-remove-osd-with-mirroring.py

```diff
@@ -0,0 +1,188 @@
+from ceph.parallel import parallel
+from tests.rbd_mirror.rbd_mirror_utils import execute_dynamic, rbd_mirror_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_add_remove_osd_with_mirroring(rbd_mirror, pool_type, **kw):
+    try:
+        mirror1 = rbd_mirror.get("mirror1")
+        mirror2 = rbd_mirror.get("mirror2")
+        config = kw.get("config")
+        pool = config[pool_type]["pool"]
+        image = config[pool_type]["image"]
+        imagespec = pool + "/" + image
+
+        results = {}
+
+        with parallel() as p:
+            p.spawn(
+                execute_dynamic,
+                mirror1,
+                "benchwrite",
+                results,
+                imagespec=imagespec,
+                io="100M",
+            )
+            p.spawn(
+                execute_dynamic,
+                mirror1,
+                "add_remove_osd",
+                results,
+                command="add",
+                osd_nodes=config["osds_to_add"],
+                config=kw.get("config"),
+            )
+
+        if results["benchwrite"] or results["add_remove_osd"]:
+            log.error("Adding OSDs failed for primary cluster")
+            return 1
+        try:
+            mirror1.check_data(mirror2, imagespec)
+        except BaseException:
+            log.error(
+                "Data consistency check failed after adding OSDs to primary cluster"
+            )
+            return 1
+
+        with parallel() as p:
+            p.spawn(
+                execute_dynamic,
+                mirror1,
+                "benchwrite",
+                results,
+                imagespec=imagespec,
+                io="100M",
+            )
+            p.spawn(
+                execute_dynamic,
+                mirror1,
+                "add_remove_osd",
+                results,
+                command="rm",
+                osd_nodes=config["osds_to_remove"],
+                config=kw.get("config"),
+            )
+
+        if results["benchwrite"] or results["add_remove_osd"]:
```

Comment:
why `benchwrite` check here?

Reply:
Because we are running IOs in parallel with OSD addition and removal, so we have to check whether both happened successfully.

Reply:
condition should be `and ` right if we are checking both then.

Reply:
@sunilangadi2
We should be failing the TC if any one of the func fails, not just only when both of them, Right ?

Reply:
Thanks @VasishtaShastry got it.